rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/util/PackageNameConverter.groovy

```diff
@@ -45,29 +45,25 @@ class PackageNameConverter {
   // For example: nflx-djangobase-enhanced_0.1-h12.170cdbd_all
   public static OsPackageName parseDebPackageName(String fullyQualifiedPackageName) {
     OsPackageName osPackageName = new OsPackageName()
+    if (!fullyQualifiedPackageName) return osPackageName
 
     osPackageName.with {
-      List<String> parts = fullyQualifiedPackageName?.tokenize("_")
-
-      if (parts) {
-        name = parts[0]
-
-        if (parts.size > 1) {
-          List<String> versionReleaseParts = parts[1].tokenize("-")
-
-          if (versionReleaseParts) {
-            version = versionReleaseParts[0]
-
-            if (versionReleaseParts.size > 1) {
-              release = versionReleaseParts[1]
-            }
-          }
-
-          if (parts.size > 2) {
-            arch = parts[2]
-          }
-        }
+      try {
+        List<String> parts = fullyQualifiedPackageName?.tokenize("_")
+        name = parts.get(0)
+
+        List<String> versionReleaseParts = parts.get(1).tokenize("-")
+        version = versionReleaseParts.get(0)
+        release = versionReleaseParts.get(1)
+        arch = parts[2]
+      } catch (RuntimeException e) {
```

Comment:
Would rather we check the explicit conditions that might be false and then declare the naming-convention violated, instead of just catching any/all exceptions that might be thrown. Can still collect the extract the contents of the`catch` block into a utility method to be used at each spot we decide the name is invalid.